tests/unit/httpd_test.cc

```diff
@@ -810,10 +810,23 @@ SEASTAR_TEST_CASE(test_client_unexpected_reply_status) {
                 }
             };
             auto cln = http::experimental::client(std::make_unique<connection_factory>(lcf));
-            auto req = http::request::make("GET", "test", "/test");
-            BOOST_REQUIRE_THROW(cln.make_request(std::move(req), [] (const http::reply& rep, input_stream<char>&& in) {
-                return make_ready_future<>();
-            }).get0(), httpd::unexpected_status_error);
+            {
+                auto req = http::request::make("GET", "test", "/test");
+                BOOST_REQUIRE_THROW(cln.make_request(std::move(req), [] (const http::reply& rep, input_stream<char>&& in) {
+                    BOOST_REQUIRE(false); // should throw before handling response
+                    return make_ready_future<>();
+                }, http::reply::status_type::ok).get0(), httpd::unexpected_status_error);
+            }
+            {
+                auto req = http::request::make("GET", "test", "/test");
+                http::reply::status_type status;
```

Comment:
I'd make it `std::optional<status_type>` and ...

Reply:
done